### PR TITLE
🌱Bump CAPI to v1.3.1

### PR DESCRIPTION
**Pull request by mdbooth**
/hold

### CODE REVIEW

File: test/e2e/suites/e2e/clusterctl_upgrade_test.go [hunk 1]
```diff
@@ -40,15 +40,16 @@ var _ = Describe("When testing clusterctl upgrades (v0.6=>current) [clusterctl-u
...
-			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.3.0/clusterctl-{OS}-{ARCH}",
-			InitWithProvidersContract: "v1beta1",
-			MgmtFlavor:                shared.FlavorDefault,
-			WorkloadFlavor:            shared.FlavorV1alpha5,
+			E2EConfig:                       e2eCtx.E2EConfig,
+			ClusterctlConfigPath:            e2eCtx.Environment.ClusterctlConfigPath,
+			BootstrapClusterProxy:           e2eCtx.Environment.BootstrapClusterProxy,
+			ArtifactFolder:                  e2eCtx.Settings.ArtifactFolder,
+			SkipCleanup:                     false,
+			InitWithBinary:                  "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.3.0/clusterctl-{OS}-{ARCH}",
```
Comment (lentzi90) [inline test/e2e/suites/e2e/clusterctl_upgrade_test.go:48]: I guess we should also bump this CAPI version, maybe even add a parameter for it in the e2e_conf.yaml.
We have it also in multiple places in the e2e_conf.yaml ([here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/54a8297ba2525dd515f264db78baf29b0a14b97a/test/e2e/data/e2e_conf.yaml#L15-L20) and [here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/54a8297ba2525dd515f264db78baf29b0a14b97a/test/e2e/data/e2e_conf.yaml#L32-L73)).

File: test/e2e/suites/e2e/clusterctl_upgrade_test.go [hunk 2]
```diff
@@ -40,15 +40,16 @@ var _ = Describe("When testing clusterctl upgrades (v0.6=>current) [clusterctl-u
...
+			InitWithProvidersContract:       "v1beta1",
+			InitWithInfrastructureProviders: []string{"openstack:v0.6.4"},
```
[the remark this answers is not in the document]
Comment (lentzi90) [reply test/e2e/suites/e2e/clusterctl_upgrade_test.go:50]: Yes!
@mdbooth this needs to match what is in the [e2e_conf.yaml](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/54a8297ba2525dd515f264db78baf29b0a14b97a/test/e2e/data/e2e_conf.yaml#L74-L91). Now it is trying to look for v0.6.4 but only v0.6.3 is in the config. You can see the repository structure that it is trying to use [here in the test artifacts](https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1424/pull-cluster-api-provider-openstack-e2e-full-test/1603195471436713984/artifacts/repository/infrastructure-openstack/).

**Review by lentzi90 — COMMENTED** [review]
Looks like it is working as intended. We just need to update the e2e_conf.yaml so it matches the test.

I noticed one warning in the test logs about ginkgo version mismatch. Not sure where it is coming from though. Do we install the ginkgo CLI somewhere with a different version than the go modules?
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1424/pull-cluster-api-provider-openstack-e2e-full-test/1603195471436713984#1:build-log.txt%3A1587

**Comment by mdbooth** [comment]
Looking good! @lentzi90 could you drop lgtm on it if you're happy?

**Review by lentzi90 — COMMENTED** [review]
Nice!
Clusterctl upgrade log looks spot on :heavy_check_mark: 
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1424/pull-cluster-api-provider-openstack-e2e-full-test/1603347053381947392/artifacts/clusters/clusterctl-upgrade-jpk36k/clusterctl-upgrade.log
/lgtm

**Pull request merged**